File: virtusbk/src/main/java/com/dua/virtusbk/entity/Course.java
package com.dua.virtusbk.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDateTime;

@Table(name = "courses")
@Data
@NoArgsConstructor
@Entity
@ToString

public class Course {
    public Course(Long id) {
        this.id = id;
    }

    @Id
    @SequenceGenerator(name = "courses_id_course_seq", sequenceName = "courses_id_course_seq", allocationSize = 1)
    @GeneratedValue(generator = "courses_id_course_seq")
    @Column(name = "id_course", nullable = false)
    private Long id;

    @Column(name = "name_course", nullable = false, length = 50)
    private String nameCourse;

    @Column(name = "description_course", nullable = false, length = 100)
    private String descriptionCourse;

    @Column(name = "keywords_course")
    @Lob
    @Type(type = "org.hibernate.type.TextType")
    private String keywordsCourse;

    @Column(name = "pathimg_course", length = 400)
    private String pathimgCourse;

    @Column(name = "datereg_course", nullable = false)
    private LocalDateTime dateregCourse;

    @Column(name = "dateupdate_course", nullable = false)
    private LocalDateTime dateupdateCourse;

    @Column(name = "state_course", nullable = false, length = 1)
    private String stateCourse;

    @Column(name = "language_course", length = 20)
    private String languageCourse;

    @Column(name = "price_course", precision = 2)
    private BigDecimal priceCourse;

    @ManyToOne
//    @Cascade(value={org.hibernate.annotations.CascadeType.ALL})
    @JoinColumn(name = "persons_id_person", nullable = false)
    //@JsonIgnore// No traer toda la relación en una consulta
    private Person personsIdPerson;

}

File: virtus/src/app/models/Course.ts
import {Person} from "./Person";

export class Course {
  private id: number;
  private nameCourse: string;
  private descriptionCourse: string;
  private keywordsCourse: string;
  private pathimgCourse: string;
  private dateregCourse: string;
  private dateupdateCourse: string;
  private stateCourse: string;
  private languageCourse: string;
  private priceCourse: string;
  private personsIdPerson: Person;


  constructor(id: number, nameCourse: string, descriptionCourse: string, keywordsCourse: string, pathimgCourse: string, dateregCourse: string, dateupdateCourse: string, stateCourse: string, languageCourse: string, priceCourse: string) {
    this.id = id;
    this.nameCourse = nameCourse;
    this.descriptionCourse = descriptionCourse;
    this.keywordsCourse = keywordsCourse;
    this.pathimgCourse = pathimgCourse;
    this.dateregCourse = dateregCourse;
    this.dateupdateCourse = dateupdateCourse;
    this.stateCourse = stateCourse;
    this.languageCourse = languageCourse;
    this.priceCourse = priceCourse;
  }

  get _id(): number {
    return this.id;
  }

  set _id(value: number) {
    this.id = value;
  }

  get _nameCourse(): string {
    return this.nameCourse;
  }

  set _nameCourse(value: string) {
    this.nameCourse = value;
  }

  get _descriptionCourse(): string {
    return this.descriptionCourse;
  }

  set _descriptionCourse(value: string) {
    this.descriptionCourse = value;
  }

  get _keywordsCourse(): string {
    return this.keywordsCourse;
  }

  set _keywordsCourse(value: string) {
    this.keywordsCourse = value;
  }

  get _pathimgCourse(): string {
    return this.pathimgCourse;
  }

  set _pathimgCourse(value: string) {
    this.pathimgCourse = value;
  }

  get _dateregCourse(): string {
    return this.dateregCourse;
  }

  set _dateregCourse(value: string) {
    this.dateregCourse = value;
  }

  get _dateupdateCourse(): string {
    return this.dateupdateCourse;
  }

  set _dateupdateCourse(value: string) {
    this.dateupdateCourse = value;
  }

  get _stateCourse(): string {
    return this.stateCourse;
  }

  set _stateCourse(value: string) {
    this.stateCourse = value;
  }

  get _languageCourse(): string {
    return this.languageCourse;
  }

  set _languageCourse(value: string) {
    this.languageCourse = value;
  }

  get _priceCourse(): string {
    return this.priceCourse;
  }

  set _priceCourse(value: string) {
    this.priceCourse = value;
  }


  get _personsIdPerson(): Person {
    return this.personsIdPerson;
  }

  set _personsIdPerson(value: Person) {
    this.personsIdPerson = value;
  }
}
